gns3/graphics_view.py

```diff
@@ -293,36 +294,39 @@ def addLinkSlot(self, link_id):
             self.deleteLinkSlot(link_id)
             return
 
-        # ugly multi-link management
-        # FIXME: taken from old GNS3 and has a bug!
-        multi = 0
-        d1 = 0
-        d2 = 1
-        link_items = source_item.links()
-        for link_item in link_items:
-            if link_item.destinationItem().node().id() == destination_item.node().id():
-                d1 += 1
-            if link_item.sourceItem().node().id() == destination_item.node().id():
-                d2 += 1
-
-        if len(link_items) > 0:
-            if d2 - d1 == 2:
-                source_port, destination_port = destination_port, source_port
-                source_item, destination_item = destination_item, source_item
-                multi = d1 + 1
-            elif d1 >= d2:
-                source_port, destination_port = destination_port, source_port
-                source_item, destination_item = destination_item, source_item
-                multi = d2
-            else:
-                multi = d1
+        # Multi-link management
+        #
+        # multi is the offset of the link
+        # +------+       multi = -1    Link 3  +-------+
+        # |      +-----------------------------+       |
+        # |  R1  |                             |   R2  |
+        # |      |        multi = 0    Link 1  |       |
+        # |      +-----------------------------+       |
+        # |      |        multi = 1    Link 2  |       |
+        # +------+-----------------------------+-------+
 
-        # MAX 7 links on the scene between 2 nodes
-        if multi > 3:
+        if source_item == destination_item:
+            multi = 0
+        else:
             multi = 0
+            link_items = source_item.links()
+            for link_item in link_items:
+                if link_item.destinationItem().node().id() == destination_item.node().id():
+                    multi += 1
+                if link_item.sourceItem().node().id() == destination_item.node().id():
+                    multi += 1
 
-        if source_item == destination_item:
+        # MAX 7 links on the scene between 2 nodes
+        if multi > 7:
             multi = 0
+        # Pair item represent the bottom links
+        elif multi % 2 == 0:
+            multi = multi / 2
+        else:
+            multi = math.ceil(float(multi) / 2) * -1
```

Comment:
A 324 it's not possible to get a float. But you are right i will rewrote it with your suggest